pkg/catalogsourceconfig/registry.go

```diff
@@ -126,7 +126,7 @@ func (r *registry) ensureDeployment(appRegistries []string, needServiceAccount b
 
 		// Wait for the deployment to scale down. We need to get the latest version of the object after
 		// the update, so we use the object returned here for scaling up.
-		if deployment, err = r.waitForDeploymentScaleDown(2*time.Second, 1*time.Minute); err != nil {
+		if _, err = r.waitForDeploymentScaleDown(2*time.Second, 1*time.Minute); err != nil {
```

Comment:
Won't you have the same issue with other resources like  `roles`, `rolebindings` etc in the scenario where the update fails?

Reply:
I don't see anything in those functions that could cause a similar issue - did I miss something?

Reply:
```
		role.Rules = getRules()
		err = r.client.Update(context.TODO(), role)
		if err != nil {
			r.log.Errorf("Failed to update Role %s : %v", role.GetName(), err)
			return err
		}
		r.log.Infof("Updated Role %s", role.GetName())
	}
```
Would an error in the update call above result in `role` becoming `nil`? If that won't happen, then the fix you have made will suffice.

Reply:
No, the issue with `waitForDeployment` is that it explicitly returns `nil, err` if there's an error (which happens when there's a timeout). The client calls won't nil out a non-nil reference that's passed in.